src/lib/layer-manager.js

```diff
@@ -200,7 +265,51 @@ export default class LayerManager {
     return redraw;
   }
 
+  //
   // PRIVATE METHODS
+  //
+
+  /**
+   * Route click events to layers.
+   * `pickLayer` will call the `onClick` prop of any picked layer,
+   * and `onLayerClick` is called directly from here
+   * with any picking info generated by `pickLayer`.
+   */
+  _onClick(event) {
+    const pos = getPos(event);
```

Comment:
So the main thing the user needs to specify is how to extract the position from the event. The rest of the logic here is independent of event handling system, right?

So exploring the idea of keeping event registration separate, we could potentially move out the event registration, keep these functions in here (maybe removing the `_` underscore prefixes) so that the can be called by the external event handlers, and add a `getPositition` argument to the `LayerManager` constructor ?